### DISCUSSION TITLE
Q: How to listen the OrderFulfilled event with ethers.js?

### DISCUSSION THREAD

**Discussion by chococrypto** [ProjectOpenSea/seaport#543] [Q&A]
I started to develop a program that tracks the portfolio when buying and selling on Opensea. To make this little program work, I based on the "OrdersMatched" event of the old contract. 

I noticed that opensea now uses the seaport protocol (this is amazing 🔥). So I have to modify my code (🤦🏽‍♂️), but I have difficulties listening to the "OrderFulfilled" event when a wallet buys 1 NFT with ETH. 

Can you guide me to do this? Is it possible to listen to this event?

**Comment by ryanio** [comment]
hey, yes it should be as simple as filtering for the topic for event `OrderFulfilled`, do you have a code example you can share for your `OrdersMatched` and can give you pointers to update it? the topic to filter for `OrderFulfilled` should be `0x9d9af8e38d66c62e2c12f0225249fd9d721c54b83f48d9352c97c6cacdcb6f31` according to https://etherscan.io/address/0x00000000006c3852cbef3e08e8df289169ede581#events

below are the event items, so `offerer` and `zone` (the indexed args) are the 2nd and 3rd topics respectively:
https://github.com/ProjectOpenSea/seaport/blob/891b5d4f52b58eb7030597fbb22dca67fd86c4c8/contracts/interfaces/ConsiderationEventsAndErrors.sol#L27-L34

related question: https://github.com/ProjectOpenSea/seaport/discussions/546

**Reply by chococrypto** [reply]
Here is the code I was developing

```
const { ethers } = require("ethers");
const {openseaContract} = require("./exchanges.js");
const abiERC721 = require('./abi/erc721-standard.json');
const { provider } = require("./provider.js");


const walletFollowed = [ ... ]; // wallet erc20
const filterEvt = seaPortContract.filters
const filterEvt = openseaContract.filters.OrdersMatched(null, null, null, walletFollowed[0]);

openseaContract.on(filterEvt, async (
  buyHash, sellHash, maker, taker, price, metadata, event
) => {
   const logs = (await event.getTransactionReceipt()).logs;
   const logsTransfer = logs[logs.length - 2];
   console.log(logsTransfer);
   if(logsTransfer) {
       const addressContractNFT = logsTransfer.address;
       const idNFT = parseInt(logsTransfer.topics[logsTransfer.topics.length - 1]);
       const nftContract = new ethers.Contract(addressContractNFT, abiERC721, provider);
       const nameNFT = await nftContract.name();
       console.log("name NFT: ", nameNFT);
   }
})
    
```

how to make us a filter on a hasher event ?
